packages/labs/ssr/README.md

```diff
@@ -51,7 +60,83 @@ const ssrResult = await (renderModule(
   [{some: "data"}]         // Arguments to function
 ) as Promise<Iterable<unknown>>);
 
-for (let text of ssrResult) {
-  console.log(text);
-}
+// ...
+
+context.body = Readable.from(ssrResult);
+```
+
+## Client usage
+
+### Hydrating lit templates
+
+In order to "hydrate" lit templates, which is the process of having lit re-associate the expressions of a lit template with which nodes they should update in the DOM, the `experimental-hydrate` module is provided in the `lit` package. Prior to updating a server-rendered container using `render`, you should first call `hydrate` on that container using the same template and data that was used to render on the server:
+
+```js
+import {myTemplate} from './my-template.js';
+import {render, hydrate} from `lit/experimental-hydrate.js`;
+
+// Initial hydration required before render:
+const initialData = /* should be same data used to render on the server */;
```

Comment:
We currently require hydrating with same data, because we don't dirty-check or re-set every attribute or textContent during hydration. That's a choice, but because we prime the dirty checking with the values from hydration, they must start off in sync with the DOM. We discussed perf testing and/or adding a switch to allow possibly different data (which would require at least a dirty check of all values during hydration), but that's not yet implemented.

Re: formatting, the idea was this is a top-level module that's loading and hydrating the page on module load, and providing an update function to update it in the future.

Reply:
Ah right, then change "should" to "must"

Reply:
As for formatting, I thought the statement was `initialData = hydrate(...)`. So maybe make an example data function:
```suggestion
/* must be same data used to render on the server */;
const initialData = getInitialAppData();
```